cfgov/cfgov/settings/base.py

```diff
@@ -12,6 +12,27 @@
 from cfgov.util import admin_emails
 
 
+# File based secrets
+_SECRETS_DIR = os.getenv("SECRETS_DIR", "/var/run/secrets/cfgov")
+if os.path.isdir(_SECRETS_DIR):
+    # Walk all levels, "exporting" filenames as environment variables
+    # and their contents as the values, for the Python/Django runtime.
+    # Trailing newline is stripped.
+    for root, dirs, files in os.walk(_SECRETS_DIR):
+        for file in files:
+            # Ignore SECRETS_FOLLOW_SYMLINKS file (security).
+            # If SECRETS_FOLLOW_SYMLINKS is set as an actual var,
+            # and it is true, then "follow" (open) the symlink.
+            # This is not secure to use in production!!!
+            if file.upper() == "SECRETS_FOLLOW_SYMLINKS" or (
+                os.getenv("SECRETS_FOLLOW_SYMLINKS", "").lower() != "true"
+                and os.path.islink(os.path.join(root, file))
+            ):
+                continue
+            with open(os.path.join(root, file)) as f:
+                os.environ[file] = f.read().strip()
+
+
```

Comment:
My main concern would be... this is a relatively large amount of logic to have in the settings file. How would you feel about breaking this logic off into a separate module that can be called when Django loads (here would be okay to be independent of how Django is started)? 

That would let us wrap it in some unittests to help self-document how it works. It would also give us a place to refactor independent of the settings file, if we ever needed to add some additional checks around values, etc. 

If this is a general pattern in Kubernetes, and [it appears to be](https://kubernetes.io/docs/concepts/configuration/secret/#uses-for-secrets), then maybe this could be an independent Python package?